tests/drivers/rpc_flash/src/main.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2020-2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/storage/flash_map.h>
+
+#define TEST_NODE		DT_NODELABEL(rpc_flash_controller)
+#define TEST_AREA_DEVICE	DEVICE_DT_GET(TEST_NODE)
+#define TEST_AREA_MAX 		DT_REG_SIZE(TEST_NODE)
+#define TEST_AREA_OFFSET	0x10000
+#define EXPECTED_SIZE		512
+#define CANARY			0xff
+
+static const struct device *const flash_dev = TEST_AREA_DEVICE;
+static struct flash_pages_info page_info;
+static uint8_t __aligned(4) expected[EXPECTED_SIZE];
+
+static void *flash_driver_setup(void)
+{
+	int rc;
+
+	zassert_true(device_is_ready(flash_dev));
+
+	const struct flash_parameters *flash_params =
+			flash_get_parameters(flash_dev);
+
+	flash_get_page_info_by_offs(flash_dev, TEST_AREA_OFFSET,
+				    &page_info);
+
+	/* Check if test region is not empty */
+	uint8_t buf[EXPECTED_SIZE];
+
+	rc = flash_read(flash_dev, TEST_AREA_OFFSET,
```

Comment:
I do not think this is correct. Here the output of the `flash_get_page_info_by_offs` should rather be used as it is not warranted that `TEST_AREA_OFFSET` will be `page_info.start_offset`.